### PR TITLE
Add sh.funkplot.Funkplot

### PR DESCRIPTION
**Pull request by fritzone**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: org.unafrog.funkplot.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+app-id: org.unafrog.funkplot
```
Comment (hfiguiere) [inline org.unafrog.funkplot.yml:1]: There doesn't seem to be a domain `unaforg.org`. So this app-id doesn't meet the requirements.

File: org.unafrog.funkplot.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+runtime: org.kde.Platform
+runtime-version: '5.15-22.08'
+sdk: org.kde.Sdk
+base: "io.qt.qtwebengine.BaseApp"
+base-version: "5.15-22.08"
+command: funkplot-gui
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=wayland
```
[the remark this answers is not in the document]
Comment (fritzone) [reply org.unafrog.funkplot.yml]: changed to `fallback-x11`

File: org.unafrog.funkplot.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  - --filesystem=host
+  - --device=dri
+modules:
+  - name: funkplot
+    buildsystem: cmake
+    config-opts:
+      - "-DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/app -DFLATPAK_BUILD=ON"
```
[the remark this answers is not in the document]
Comment (fritzone) [reply org.unafrog.funkplot.yml]: done

File: org.unafrog.funkplot.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+command: funkplot-gui
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=wayland
+  - --filesystem=host
+  - --device=dri
+modules:
+  - name: funkplot
+    buildsystem: cmake
```
[the remark this answers is not in the document]
Comment (fritzone) [reply org.unafrog.funkplot.yml]: done

**Comment by hfiguiere** [comment]
bot, build sh.funkplot.Funkplot

**Comment by hfiguiere** [comment]
it's missing the required appstream file, and the .desktop and icons have to be renamed.

File: sh.funkplot.Funkplot.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+runtime-version: '5.15-22.08'
+sdk: org.kde.Sdk
+base: "io.qt.qtwebengine.BaseApp"
+base-version: "5.15-22.08"
+command: funkplot-gui
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --filesystem=home
+  - --device=dri
```
Comment (hfiguiere) [inline sh.funkplot.Funkplot.yml:12]: also missing `cleanup` rules to remove files not needed at build time like the headers and the cmake files.

File: sh.funkplot.Funkplot.yml [hunk 6]
```diff
@@ -0,0 +1,22 @@
+app-id: sh.funkplot.Funkplot
+runtime: org.kde.Platform
+runtime-version: '5.15-22.08'
+sdk: org.kde.Sdk
+base: "io.qt.qtwebengine.BaseApp"
+base-version: "5.15-22.08"
+command: funkplot-gui
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
```
Comment (hfiguiere) [inline sh.funkplot.Funkplot.yml:10]: this won't work. I didn't say to remove the wayland socket if it support wayland. Nor to change to fallback-x11 if it doesn't.

**Comment by fritzone** [comment]
Sorry, this is all new to me, I just used the example provided at https://docs.flatpak.org/en/latest/qt.html for Qt applications, tailored it to suit my needs and hoped that it will work out of the box. When I will have a bit more time I will invest some energy in properly creating a flatpak package.

**Comment by hfiguiere** [comment]
bot, build sh.funkplot.Funkplot

**Comment by hfiguiere** [comment]
> it's missing the required appstream file, and the .desktop and icons have to be renamed.

Whithout this the build will never be successful.

**Pull request closed without merging**